app/controllers/pet_applications_controller.rb

```diff
@@ -10,14 +10,14 @@ def update
     if params[:approve_all]
       applicant.pets.each do |pet|
         pet.update(pet_params)
-        pet.applicants = []
+        pet.applicants.delete(applicant)
```

Comment:
Very interesting here to delete the single applicant and then reorder it to access the correct applicant at the end!